tests/test_labelled_pandas.py

```diff
@@ -34,292 +35,329 @@ def assert_frame_equal_not_index(x, y):
 
 
 def test_LabelledSeries():
-    index = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J']
-    labels = ['$%s$' % i for i in index]
-    data = np.random.rand(len(index))
+    with warnings.catch_warnings():
+        warnings.filterwarnings(
+            "ignore",
+            message=".*indexing past lexsort depth may impact performance.*",
+        )
```

Comment:
I think a function decorator should work, here, and then we don't need the extra indent (will also make it much easier to review here on GitHub...).

```python
@pytest.mark.filterwarnings("ignore::PerformanceWarning")
def test_LabelledSeries():
    ...
```

Same for the other tests below.

Reply:
I get the following error:
```
ERROR: while parsing the following warning configuration:

  ignore::PerformanceWarning

This error occurred:

Traceback (most recent call last):
  File "/Users/adam/Programming/env_oct03/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1639, in parse_warning_filter
    category: Type[Warning] = _resolve_warning_category(category_)
  File "/Users/adam/Programming/env_oct03/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1678, in _resolve_warning_category
    cat = getattr(m, klass)
AttributeError: module 'builtins' has no attribute 'PerformanceWarning'
```

Any ideas how to solve?

Reply:
This is a pandas warning, so it might need importing at the head of the file.
```python
from pandas.errors import PerformanceWarning
```

Reply:
Try:
```python
@pytest.mark.filterwarnings("ignore::pandas.errors.PerformanceWarning")
```

Reply:
perfect

Reply:
Ah, so much nicer to review now, too :)